### PR TITLE
feature: add-memcached-for-friendships-api

### PR DESCRIPTION
**Pull request by W-Shih**
- install: install python-memcached
- settings: set up cache
- feature: add-Memcached-for-friendships-API with Django signal-listener
- test: add a test for FriendshipService

### CODE REVIEW

File: friendships/tests.py [hunk 1]
```diff
@@ -106,3 +109,28 @@ def test_friendship_str(self):
...
+        kd35_following_id_set = set()
+        for i in range(3):
+            user = self.create_user(username=f'kd35_following:{i}')
+            kd35_following_id_set.add(user.id)
+            Friendship.objects.create(from_user=self.kd35, to_user=user)
+        Friendship.objects.create(from_user=self.kd35, to_user=self.mj23)
+        kd35_following_id_set.add(self.mj23.id)
+
+        following_user_id_set = FriendshipService.get_following_user_id_set(self.kd35.id)
+        self.assertSetEqual(following_user_id_set, kd35_following_id_set)
```
Comment (wen-wayne-shih) [inline friendships/tests.py:131]: should call get_following_user_id_set() twice to test cache.
Get followings from DB and set to cache first time, and get followings from cache second time

File: friendships/services.py [hunk 2]
```diff
@@ -72,5 +79,20 @@ def get_followers(cls, user_id):
         return followers
 
     @classmethod
-    def get_has_followed(cls, from_user, to_user):
-        return Friendship.objects.filter(from_user=from_user, to_user=to_user).exists()
+    def get_following_user_id_set(cls, from_user_id):
+        key = FOLLOWINGS_PATTERN.format(user_id=from_user_id)
+        following_user_id_set = cache.get(key)
+        if following_user_id_set is not None:
+            return following_user_id_set
```
Comment (wen-wayne-shih) [inline friendships/services.py:86]: should call get_following_user_id_set() twice to test cache.
Get followings from DB and set to cache first time, and get followings from cache second time

**Review by wen-wayne-shih — CHANGES_REQUESTED** [review]
Please resolve the review comments

File: friendships/tests.py (hunk 1 above)
Comment (W-Shih) [reply friendships/tests.py:132]: Resolved as suggestion

File: friendships/services.py (hunk 2 above)
Comment (W-Shih) [reply friendships/services.py:86]: Resolved as suggestion

**Review by wen-wayne-shih — APPROVED** [review]
LGTM, squash before merging

**Pull request merged**